src/tests/test_config.cpp

```diff
@@ -404,4 +403,38 @@ BOOST_DATA_TEST_CASE(add_child_NonEmptyThis_newOrExistingKey_lOrRValue_AppendAnd
 	BOOST_CHECK_NE(&new_child, &update);
 }
 
+BOOST_AUTO_TEST_CASE(auto_conversion_bool_and_int)
+{
+	config cfg;
+	cfg["boolAttributeTrue"] = true;
```

Comment:
Please use snake case.

Reply:
In other words:
```suggestion
	cfg["bool_attribute_true"] = true;
```
(I have no idea what the reasoning for this is, but I personally don't care one way or the other.)

Reply:
It's just for consistency. Nearly all of wesnoth's code is in snake case, aside from a few bits of ancient code that have stuck around.

Reply:
Well, it's debatable whether a string key in a config counts as code, but sure.